### PR TITLE
Added Transaction Type

### PR DESCRIPTION
**Pull request by EToreo**
Pull request for https://github.com/sequelize/sequelize/issues/4901

### CODE REVIEW

**Comment by janmeier** [comment]
LGTM, just needs some tests and jshint-fixes

**Comment by mickhansen** [comment]
PG has a deferred mode aswell that we support in a different way, not sure if its comparable to SQLITE dererred transaction?

**Comment by EToreo** [comment]
@janmeier I wasn't sure what to do about tests. Should I just check to make sure the correct syntax is output based on the different options? Seems like a lame test. 

@mickhansen PG has a few options in a similar vein, but they can be set inside the transaction, where as these must be set as part of the start of the transaction, that is why I felt it warranted a new set of options. MySQL and I think PG also support a read only mode, but I think that belongs in a different set of options for itself as well.

**Comment by EToreo** [comment]
Added tests and corrected lint.

**Comment by EToreo** [comment]
Can I get this pulled in or suggestions on improvements before a pull is granted?

**Comment by mickhansen** [comment]
I'm a bit on the fence on branching out the options so much, if PG supports DEFERRED aswell we could look to provide a standardized syntax. 

PG has INITIALLY_DEFERRED/INITIALLY_IMMEDIATE, would that map to your DEFERRED/IMMEDIATE here?

**Comment by EToreo** [comment]
@mickhansen I can see where you are coming from.  But here is where I was coming from... PG calls out their options as isolation levels AND allows that setting to be changed after a BEGIN statement, while Sqlite calls their setting a behavior (and can only be set with the BEGIN statement).

If all these are really isolation levels, then let's wrap them all up.  If not, I like the current implementation.

Thoughts?

**Comment by mickhansen** [comment]
@EToreo No idea if they are actually equal.
thoughts @janmeier?

I suppose having a type of transaction makes sense, even for PG (we could later support types for PG aswell with the INITIALLY_DEFERRED)

**Comment by EToreo** [comment]
@janmeier I know this is a busy time of year, but have you had a chance to look at this and chime in?

I'm looking to get it pulled in for a project I am working on and this is becoming gating.

**Comment by janmeier** [comment]
I think there is a difference here, since the pg deferred stuff can also be limited to specific tables etc - I like your implementation here, even though its sqlite specific

**Comment by EToreo** [comment]
Wonderful.  Let me know if there is anything more I can do to get this pulled in.

File: test/integration/transaction.test.js [hunk 1]
```diff
@@ -239,6 +239,32 @@ describe(Support.getTestDialectTeaser('Transaction'), function() {
...
+  if (current.dialect.supports.transactionOptions.type) {
+    describe('transaction types', function() {
+      it('should support default transaction type DEFERRED', function() {
+        return this.sequelize.transaction({
+        }).bind(this).then(function (t) {
+          return t.rollback().bind(this).then(function() {
+            expect(t.options.type).to.equal('DEFERRED');
+          });
+        });
+      });
```
Comment (mickhansen) [inline test/integration/transaction.test.js]: `DEFERRED` is the default for sqlite if none is defined?

Comment (EToreo) [inline test/integration/transaction.test.js]: Yep. 
> Transactions can be deferred, immediate, or exclusive. The default transaction behavior is deferred.

https://www.sqlite.org/lang_transaction.html

File: lib/sequelize.js [hunk 2]
```diff
@@ -1200,6 +1202,7 @@ Sequelize.where = Sequelize.condition = Sequelize.prototype.condition = Sequeliz
 
  * @param {Object}   [options={}]
  * @param {Boolean}  [options.autocommit=true]
+ * @param {String}   [options.type='DEFERRED'] See `Sequelize.Transaction.TYPES` for possible options
```
Comment (mickhansen) [inline lib/sequelize.js]: Might want to add a sqlite only notice here and the other place you mention it in the docs.

**Comment by mickhansen** [comment]
Just a single comment then i think we're good to go

**Comment by EToreo** [comment]
Good suggestion.  Notes added.

**Comment by EToreo** [comment]
@mickhansen  @janmeier Thanks guys.
Can you give your thoughts and input on https://github.com/sequelize/sequelize/pull/4969 now that this perquisite has been satisfied?

**Pull request merged**